lib/improver/argparser.py

```diff
@@ -28,21 +28,190 @@
-    """Argument parser for improver CLIs"""
+class ArgParser(ArgumentParser):
+    """Argument parser for improver CLIs.
+
+    The main purpose of this class is to make it easier to create CLIs which
+    have arguments which are selected from centralized collections.
+
+    To fulfil these requirements, we define 2 class level dictionaries,
+    ArgParser.CENTRALIZED_ARGUMENTS, and ArgParser.COMPULSORY_ARGUMENTS.
+
+    For these dictionaries, each element has:
+        - a key, which is a string representing the argument name - used
+          internally to refer to a particular argument (which, in the case of
+          the CENTRALIZED_ARUGMENTS may be selected from when creating an
+          instance of the ArgParser)
+        - a value, which is a list containing 2 elements:
+           1. a list of strings containing the different flags which are
+              associated with the argument (ie.: the first argument to the
+              add_arguments() method, e.g: ['--profile', '-p'])
+           2. a dictionary containing all of the kwargs which are passed
+              to the add_argument() method (e.g:
+              {'action': 'store_true', 'default': False, 'help': ... })
+
+    The CENTRALIZED_ARGUMENTS will be selected from, as necessary, for each
+    of the CLIs that we create, and the COMPULSORY_ARGUMENTS will be
+    automatically added to all CLIs (with no option to exclude them).
+    """
+
+    # Ideally, all CLIs should select something from this dictionary:
+    # NB: --help included by default with ArgumentParser
+    CENTRALIZED_ARGUMENTS = {
+        'input_file': (
+            ['input_filepath'],
+            {'metavar': 'INPUT_FILE',
+             'help': 'A path to an input NetCDF file to be processed'}),
+        'output_file': (
+            ['output_filepath'],
+            {'metavar': 'OUTPUT_FILE',
+             'help': 'The output path for the processed NetCDF'}),
+    }
+
+    # *All* CLIs will use the options here (no option to disable them):
+    COMPULSORY_ARGUMENTS = {
+        # TODO: Implement (some of?) these:
+        #        'debug': (
+        #            ['--debug'],
+        #            {'action': 'store_true'}),
+        #        'dry_run': (
+        #            ['--dry-run'],
+        #            {'action': 'store_true'}),
+        #        'profile' : (
+        #            ['--profile'],
+        #            {'action': 'store_true'}),
+        #        'verbose': (
+        #            ['--verbose', '-v'],
+        #            {'action': 'count'}),
+        #        'version': (
+        #            ['--version'],
+        #            {'action': 'version',
+        #             'version': improver.version_message() }),
+    }
+
+    # We can override including these, but options common to everything should
+    # be in a list here:
+    # DEFAULT_CENTRALIZED_ARG_NAMES = ('input_file', 'output_file')
+    DEFAULT_CENTRALIZED_ARG_NAMES = ()
```

Comment:
These class attributes show up in the sphinx documentation, which is useful, but probably means you should put some more of the comments in the docstring rather than as comments so they show up in the documentation. We especially need something about the DEFAULT_CENTRALIZED_ARG_NAMES as this isn't described in the docstring otherwise.